lib/src/sort.dart

```diff
@@ -0,0 +1,213 @@
+    _inputs[j] = newValB;
+
+    for (var k = 0; k < _inputs.length; k++) {
+      _outputs.add(addOutput('swapped$k', width: _inputs[k].width));
+      _outputs[k] <= _inputs[k];
+    }
+  }
+}
+
+class _BitonicMerge extends Module {
+  /// A list of [Logic] that hold inputs.
+  List<Logic> _inputs = [];
+
+  /// A list of [Logic] that hold the final outputs of List of result.
+  final List<Logic> _outputs = [];
+
+  /// The [sorted] result.
+  List<Logic> get sorted => UnmodifiableListView(_outputs);
+
+  /// Merge and sort [bitonicSequence] into [isAscending] given.
+  ///
+  /// List of [Logic] will compare and swap [Logic] position based on the
+  /// [isAscending] given by [BitonicSort] to first created a bitonic sequence.
+  /// The final stage will sort the bitonic sequence into sorted order
+  /// of [isAscending].
+  _BitonicMerge(
+    Logic clk,
+    Logic reset, {
+    required bool isAscending,
+    required Iterable<Logic> bitonicSequence,
+    super.name = 'bitonic_merge',
+  }) {
+    clk = addInput('clk', clk);
+    reset = addInput('reset', reset);
+
+    for (var i = 0; i < bitonicSequence.length; i++) {
+      _inputs.add(addInput('bitonicSequence$i', bitonicSequence.elementAt(i),
+          width: bitonicSequence.elementAt(i).width));
+    }
+
+    if (_inputs.length > 1) {
+      for (var i = 0; i < 0 + _inputs.length ~/ 2; i++) {
+        final indexA = i;
+        final indexB = i + _inputs.length ~/ 2;
+        final swap = _CompareSwap(clk, reset, _inputs, indexA, indexB,
+            isAscending: isAscending);
+        _inputs = swap.swapped;
+      }
+
+      final mergeLeft = _BitonicMerge(
+        clk,
+        reset,
+        bitonicSequence: _inputs.getRange(0, _inputs.length ~/ 2),
+        isAscending: isAscending,
+        name: 'merge_left',
+      );
+      final mergeRight = _BitonicMerge(
+        clk,
+        reset,
+        bitonicSequence: _inputs.getRange(_inputs.length ~/ 2, _inputs.length),
+        isAscending: isAscending,
+        name: 'merge_right',
+      );
+
+      final mergeRes = mergeLeft.sorted + mergeRight.sorted;
+
+      for (var i = 0; i < mergeRes.length; i++) {
+        _outputs.add(addOutput('sorted_$i', width: mergeRes[i].width));
+        _outputs[i] <= mergeRes[i];
+      }
+    } else if (_inputs.length == 1) {
+      _outputs.add(addOutput('sorted_0', width: _inputs[0].width));
+      _outputs[0] <= _inputs[0];
+    }
+  }
+}
+
+/// Sort [inputs] to specified order.
+class BitonicSort extends _Sort<BitonicSort> {
+  /// The list of inputs port.
+  final List<Logic> _inputs = [];
+
+  /// The list of outputs port.
+  final List<Logic> _outputs = [];
+
+  /// The [sorted] result.
+  List<Logic> get sorted => UnmodifiableListView(_outputs);
+
+  /// Constructs a [Module] to sort list of [Logic].
+  ///
+  /// The sorting module will recursively split inputs into a bitonic sequence
+  /// and passed to [_BitonicMerge] to perform merging process and return
```

Comment:
Probably shouldn't reference private classes in a public API.  The interesting thing is that it uses the bitonic sort algorithm.